SU2_CFD/src/variables/CNEMOEulerVariable.cpp

```diff
@@ -0,0 +1,1466 @@
+ *
+ * Copyright 2012-2019, Francisco D. Palacios, Thomas D. Economon,
+ *                      Tim Albring, and the SU2 contributors.
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../include/variables/CNEMOEulerVariable.hpp"
+#include <math.h>
+
+CNEMOEulerVariable::CNEMOEulerVariable(unsigned long npoint,
+                                       unsigned long val_ndim,
+                                       unsigned long val_nvar,
+                                       unsigned long val_nprimvar,
+                                       unsigned long val_nprimvargrad,
+                                       CConfig *config) : CVariable(npoint,
+                                                                    val_ndim,
+                                                                    val_nvar,
+                                                                    config),
+                                       Gradient_Reconstruction(config->GetReconstructionGradientRequired() ? Gradient_Aux : Gradient) {
+
+  nDim         = val_ndim;
+  nVar         = val_nvar;
+
+  nPrimVar     = val_nprimvar;
+  nPrimVarGrad = val_nprimvargrad;
+
+  nSpecies     = config->GetnSpecies();
+  ionization   = config->GetIonization();
+
+  /*--- Array initialization ---*/
+  Limiter.resize(nPoint,nVar) = su2double(0.0);
+  Primitive.resize(nPoint,nPrimVar) = su2double(0.0);
+  Primitive_Aux.resize(nPoint,nPrimVar) = su2double(0.0);
+
+  Gradient_Primitive.resize(nPoint,nPrimVarGrad,nDim,0.0);
+
+  Gradient.resize(nPoint,nVar,nDim,0.0);
+
+  //cat apagar isto se só for necessario no constructor já a seguir. same para o header disto
+
+  if (config->GetReconstructionGradientRequired()) {
+    Gradient_Aux.resize(nPoint,nVar,nDim,0.0);
+  }
+
+  //cat
+
+  /*--- Define structure of the primtive variable vector ---*/
+  // Primitive: [rho1, ..., rhoNs, T, Tve, u, v, w, P, rho, h, a, rhoCvtr, rhoCvve]^T
+  // GradPrim:  [rho1, ..., rhoNs, T, Tve, u, v, w, P]^T
+  RHOS_INDEX    = 0;
+  T_INDEX       = nSpecies;
+  TVE_INDEX     = nSpecies+1;
+  VEL_INDEX     = nSpecies+2;
+  P_INDEX       = nSpecies+nDim+2;
+  RHO_INDEX     = nSpecies+nDim+3;
+  H_INDEX       = nSpecies+nDim+4;
+  A_INDEX       = nSpecies+nDim+5;
+  RHOCVTR_INDEX = nSpecies+nDim+6;
+  RHOCVVE_INDEX = nSpecies+nDim+7;
+

```

Comment:
Def. please consider encapsulating this logic somewhere, it is repeated in the next constructor.